appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -2566,30 +2525,28 @@ private Boolean getBooleanProperty(String value, String defaultValue) {
     }
 
 
-    private Integer getIntegerProperty(String value, Integer defaultValue) {
+    private static int getIntegerProperty(String value, int defaultValue) {
         String property;
         property = System.getProperty(value);
         if (property == null) {
             property = System.getenv(value.replace('.', '_'));
         }
         if (property == null) {
             return defaultValue;
-        } else {
-            return Integer.decode(property);
         }
+        return Integer.decode(property).intValue();
     }
 
-    private Long getLongProperty(String value, Long defaultValue) {
+    private static long getLongProperty(String value, long defaultValue) {
```

Comment:
Why have you made all of these get methods static? This just seems like a change rather than an improvement.

Reply:
The principle of least surprise. A member method (even private) can have dependencies and side effects to the instance that I would need check and consider. A static method cannot have those, hence will not surprise me in that way. Secondly it makes clear that the method might not be best located in this class as it does not depend on the class as a context. Helps to identify code that potentially should be in another class (one of its arguments) or in a utility to allow reuse without causing spaghetti dependencies. Sure its no big deal here, it is more of a thing that if you keep doing it the codebase get easier to reason about and less entangled.